removes the nar'sie skybox and summoned mobs when nar'sie is removed

RustingWithYou opened the pull request:
🆑 RustingWithYou
tweak: banishing nar'sie now deletes all simplemobs in the cult faction - constructs, creatures, faithless, etc.
tweak: banishing nar'sie now resets the skybox to its normal state
tweak: banishing nar'sie now properly checks for cultists when doing the global message
tweak: cult walls and floors can now be deconverted through nullrod application. banishing nar'sie will also deconvert those turfs
/🆑

Ran some tests on this on a private server, it seemed strange that sealing the HELL RIFT OF DOOM didn't get rid of the various cult mobs, and that it left the sky as a gaping hellhole of the blood god. Now if you do manage to banish Nar'Sie, you can really feel like you did a good job. Well done, you.

PsiOmegaDelta commented on code/game/gamemodes/cult/runes.dm:
```diff
@@ -822,15 +822,19 @@
+		SSskybox.change_skybox("dyable", new_use_stars = TRUE, new_use_overmap_details = TRUE)
+		for(var/mob/living/simple_animal/S in GLOB.living_mob_list_)
+			if(S.faction == "cult")
+				qdel(S)
```
Better would be 
```
if(GLOB.universe.type == /datum/universal_state/hell)
    SetUniversalState(/datum/universal_state)
```
And move the code above into `/datum/universal_state/hell/OnExit()`

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
\ No newline at end of file
+	..()
+
+/turf/simulated/wall/cult/attackby(var/obj/item/I, var/mob/user)
```
```suggestion
/turf/simulated/wall/cult/attackby(obj/item/I, mob/user)
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+
+/turf/simulated/wall/cult/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/weapon/nullrod))
+		user.visible_message("<span class='notice'>\The [user] touches \the [src] with \the [I], and it shifts.</span>", "<span class='notice'>You touch \the [src] with \the [I], and it shifts.</span>")
```
```suggestion
		user.visible_message(
			SPAN_NOTICE("\The [user] touches \the [src] with \the [I], and it shifts."),
			SPAN_NOTICE("You touch \the [src] with \the [I], and it shifts.")
		)
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+/turf/simulated/wall/cult/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/weapon/nullrod))
+		user.visible_message("<span class='notice'>\The [user] touches \the [src] with \the [I], and it shifts.</span>", "<span class='notice'>You touch \the [src] with \the [I], and it shifts.</span>")
+		src.decultify_wall()
```
```suggestion
		decultify_wall()
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+		return
+	..()
+
+/turf/simulated/floor/cult/attackby(var/obj/item/I, var/mob/user)
```
```suggestion
/turf/simulated/floor/cult/attackby(obj/item/I, mob/user)
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+
+/turf/simulated/floor/cult/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/weapon/nullrod))
+		user.visible_message("<span class='notice'>\The [user] touches \the [src] with \the [I], and it shifts.</span>", "<span class='notice'>You touch \the [src] with \the [I], and it shifts.</span>")
```
```suggestion
		user.visible_message(
			SPAN_NOTICE("\The [user] touches \the [src] with \the [I], and it shifts."),
			SPAN_NOTICE("You touch \the [src] with \the [I], and it shifts.")
		)
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+/turf/simulated/floor/cult/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/weapon/nullrod))
+		user.visible_message("<span class='notice'>\The [user] touches \the [src] with \the [I], and it shifts.</span>", "<span class='notice'>You touch \the [src] with \the [I], and it shifts.</span>")
+		src.decultify_floor()
```
```suggestion
		decultify_floor()
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+		src.decultify_floor()
+		return
+	..()
+/turf/proc/decultify_wall()
```
```suggestion

/turf/proc/decultify_wall()
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+	..()
+
+/turf/simulated/wall/cult/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/weapon/nullrod))
```
```suggestion
	if (istype(I, /obj/item/weapon/nullrod))
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+	..()
+
+/turf/simulated/floor/cult/attackby(var/obj/item/I, var/mob/user)
+	if(istype(I, /obj/item/weapon/nullrod))
```
```suggestion
	if (istype(I, /obj/item/weapon/nullrod))
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+	..()
+/turf/proc/decultify_wall()
+	var/turf/simulated/wall/cult/wall = src
+	if(!istype(wall))
```
```suggestion
	if (!istype(wall))
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+	var/turf/simulated/wall/cult/wall = src
+	if(!istype(wall))
+		return
+	if(wall.reinf_material)
```
```suggestion
	if (wall.reinf_material)
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+
+/turf/proc/decultify_floor()
+	var/turf/simulated/floor/cult/floor = src
+	if(!istype(floor))
```
```suggestion
	if (!istype(floor))
```

Spookerton commented on code/game/gamemodes/cult/cultify/de-cultify.dm:
```diff
@@ -3,4 +3,33 @@
+	if(!istype(floor))
+		return
+	else
+		ChangeTurf(/turf/simulated/floor/plating)
```
```suggestion
		ChangeTurf(/turf/simulated/floor/plating)

```

Spookerton commented on code/game/gamemodes/cult/hell_universe.dm:
```diff
@@ -31,6 +31,20 @@ In short:
 
+/datum/universal_state/hell/OnExit()
+	SSskybox.change_skybox("dyable", new_use_stars = TRUE, new_use_overmap_details = TRUE)
+	for(var/mob/living/simple_animal/S in GLOB.living_mob_list_)
```
```suggestion
	for (var/mob/living/simple_animal/S in GLOB.living_mob_list_)
```

Spookerton commented on code/game/gamemodes/cult/hell_universe.dm:
```diff
@@ -31,6 +31,20 @@ In short:
+/datum/universal_state/hell/OnExit()
+	SSskybox.change_skybox("dyable", new_use_stars = TRUE, new_use_overmap_details = TRUE)
+	for(var/mob/living/simple_animal/S in GLOB.living_mob_list_)
+		if(S.faction == "cult")
```
```suggestion
		if (S.faction == "cult")
```

Spookerton commented on code/game/gamemodes/cult/hell_universe.dm:
```diff
@@ -31,6 +31,20 @@ In short:
+		if(S.faction == "cult")
+			to_chat(S, "<span class='cult'>You hear a terrible scream from a place beyond reality, as a life's sacrifice banishes your master. The dark power animating your form wavers, and withdraws, leaving it an unliving shell of meat. Your mind puffs away, like mist under a hot sun.</span>")
+			S.set_stat(DEAD)
+	for(var/turf/simulated/wall/cult/wall)
```
```suggestion
	for (var/turf/simulated/wall/cult/wall)
```

Spookerton commented on code/game/gamemodes/cult/hell_universe.dm:
```diff
@@ -31,6 +31,20 @@ In short:
+			S.set_stat(DEAD)
+	for(var/turf/simulated/wall/cult/wall)
+		wall.decultify_wall()
+	for(var/turf/simulated/floor/cult/floor)
```
```suggestion
	for (var/turf/simulated/floor/cult/floor)
```

Spookerton commented on code/game/gamemodes/cult/hell_universe.dm:
```diff
@@ -31,6 +31,20 @@ In short:
+		wall.decultify_wall()
+	for(var/turf/simulated/floor/cult/floor)
+		floor.decultify_floor()
+	for(var/obj/effect/gateway/active/cult/G)
```
```suggestion
	for (var/obj/effect/gateway/active/cult/G)
```

Spookerton commented on code/game/gamemodes/cult/hell_universe.dm:
```diff
@@ -31,6 +31,20 @@ In short:
+	SSskybox.change_skybox("dyable", new_use_stars = TRUE, new_use_overmap_details = TRUE)
+	for(var/mob/living/simple_animal/S in GLOB.living_mob_list_)
+		if(S.faction == "cult")
+			to_chat(S, "<span class='cult'>You hear a terrible scream from a place beyond reality, as a life's sacrifice banishes your master. The dark power animating your form wavers, and withdraws, leaving it an unliving shell of meat. Your mind puffs away, like mist under a hot sun.</span>")
```
```suggestion
			to_chat(S, SPAN_OCCULT("You hear a terrible scream from a place beyond reality, as a life's sacrifice banishes your master. The dark power animating your form wavers, and withdraws, leaving it an unliving shell of meat. Your mind puffs away, like mist under a hot sun."))
```

Spookerton commented on code/game/gamemodes/cult/runes.dm:
```diff
@@ -822,16 +822,18 @@
 		for(var/mob/M in GLOB.living_mob_list_)
-			if(iscultist(M))
-				to_chat(M, "You see a vision of \the [user] keeling over dead, his blood glowing blue as it escapes \his body and dissipates into thin air; you hear an otherwordly scream and feel that a great disaster has just been averted.")
+			if(!iscultist(M))
```
```suggestion
			if (!iscultist(M))
```

Spookerton commented on code/game/gamemodes/cult/runes.dm:
```diff
@@ -822,16 +822,18 @@
-			if(iscultist(M))
-				to_chat(M, "You see a vision of \the [user] keeling over dead, his blood glowing blue as it escapes \his body and dissipates into thin air; you hear an otherwordly scream and feel that a great disaster has just been averted.")
+			if(!iscultist(M))
+				to_chat(M, "<span class='cult'>You see a vision of \the [user] keeling over dead, their blood glowing blue as it escapes their body and dissipates into thin air; you hear an otherwordly scream and feel that a great disaster has just been averted.</span>")
```
```suggestion
				to_chat(M, SPAN_OCCULT("You see a vision of \the [user] keeling over dead, their blood glowing blue as it escapes their body and dissipates into thin air; you hear an otherwordly scream and feel that a great disaster has just been averted."))
```

Spookerton commented on code/game/gamemodes/cult/runes.dm:
```diff
@@ -822,16 +822,18 @@
+				to_chat(M, "<span class='cult'>You see a vision of \the [user] keeling over dead, their blood glowing blue as it escapes their body and dissipates into thin air; you hear an otherwordly scream and feel that a great disaster has just been averted.</span>")
 			else
-				to_chat(M, "You see a vision of [name] keeling over dead, his blood glowing blue as it escapes his body and dissipates into thin air; you hear an otherwordly scream and feel very weak for a moment.")
+				to_chat(M, "<span class='cult'>You see a vision of [name] keeling over dead, their blood glowing blue as it escapes their body and dissipates into thin air; you hear an otherwordly scream and feel very weak for a moment.</span>")
```
```suggestion
				to_chat(M, SPAN_OCCULT("You see a vision of [name] keeling over dead, their blood glowing blue as it escapes their body and dissipates into thin air; you hear an otherwordly scream and feel very weak for a moment."))
```

Spookerton commented on code/game/gamemodes/cult/runes.dm:
```diff
@@ -822,16 +822,18 @@
 		qdel(src)
-		return
+
+		if(GLOB.universe.type == /datum/universal_state/hell)
```
```suggestion
		if (GLOB.universe.type == /datum/universal_state/hell)
```

Spookerton commented:
All but one of the change requests I made was ignored, I assume because they were separate and you only merged one instead of doing a batch. Runs okay, but I'd prefer the style done.

RustingWithYou committed:
banishing nar nar now banishes affiliated simplemobs


pronouns + edgy narration + fix antag status checks


removes edgy narration as it failed EVERY CHECK


moves changes to universal state, adds decultifying walls


Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/runes.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/runes.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/runes.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/runes.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/hell_universe.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/hell_universe.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/cultify/de-cultify.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/hell_universe.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/hell_universe.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/hell_universe.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
Update code/game/gamemodes/cult/hell_universe.dm

Co-authored-by: Spookerton <spkrtn@pm.me>
